### ISSUE TITLE
Stack 1.0.2 fails to parse stack.yaml files with empty flags (windows)

### ISSUE THREAD

**Issue by enolan** [commercialhaskell/stack#1672]
```
$ stack setup
Could not parse 'C:\Users\Echo\Desktop\temp\stack.yaml':
AesonException "Error in $.flags: failed to parse field 'flags': failed to parse field flags: expected HashMap Text a, encountered Null"
See http://docs.haskellstack.org/en/stable/yaml_configuration.html.
```

You can replicate this by running `stack new temp`, replacing the stack.yaml with [this](https://gist.github.com/enolan/b8511cba837a89b9628d), and running `stack setup`. Whether or not empty flag sections should be valid is debatable, but they worked in 1.0.0.

**Comment by mgsloan** [comment]
I'm guessing this is due to behavior changes in `aeson-0.10`.  Lets see what happens when lts-5 gets released next week with `aeson-0.9`, and stack starts using those snapshots.

**Comment by sjakobi** [comment]
I can't reproduce the issue with stack-1.1.2 on my machine. Please reopen if this is still a problem.
